### DISCUSSION TITLE
Rev 21 - No PRIMENET or EMACS on dist tapes? Or am I dumb?

### DISCUSSION THREAD

**Discussion by SirNibble** [prirun/p50em#16] [General]
Doing a clean install of Rev21 - is it missing PRIMENET & EMACS? 

Followed https://sysovl.info/reference_prime_drb_installing_primos210.html with the associated tapes, but there's no PRIMENET* or EMACS directory?

```
<PIP01>MFD (ALL access)

8 Files.

BOOT             BOOT_RUN_FILE_TREENAME            DYNBSP
FIX_DISK.SAVE    MAGRST           MAGSAV           MAKE.SAVE
PIP01

44 Directories.

BASIC            BATCHQ           BOOTRUN          BRMS
CMDNC0           DIAG             DOS              DOWN_LINE_LOAD*
DSM*             FORMS*_TEMPLATE  FTN              HELP*
INFO21.0         INFO21.0.1       INFO21.0.2       INFO21.0.3
INFO21.0.4       INFO21.0.5       INFO21.0.6       INFORMATION
ISYS             LIB              LIBRARIES*       MFD
NETWORK_MGT*     PRIRUN           RJESRC           RJSPLQ*
SAD              SEARCH_RULES*    SEG              SEGRUN*
SERVERS*         SPOOL*           SPOOL_DATA*      SPOOL_QUEUE*
SYSCOM           SYSOVL           SYSTEM           SYSTEM_DEBUG*
T&MRUN           TOOLS            TRANSLATORS      UP_LINE_DUMP*

```

Compared to the example 21 system

```
ld -nw

<SYS210>MFD (ALL access)

4 Files.

BOOT             BOOT_RUN_FILE_TREENAME            DYNBSP
SYS210

48 Directories.

BASIC            BATCHQ           BOOTRUN          BRMS
CMDNC0           DOS              DOWN_LINE_LOAD*  DSM*
EMACS*           FORMS*_TEMPLATE  FTN              GUEST21
HELP*            INDEX21          INFO21.0         INFO21.0.1
INFO21.0.2       INFO21.0.3       INFO21.0.4       INFO21.0.5
INFO21.0.6       INFORMATION      INVENTORY*       ISYS
JIMMY            LIB              LIBRARIES*       MFD
NETWORK_MGT*     PRIMENET*        PRIMENET*.OLD    PRIRUN
RJESRC           RJSPLQ*          SAD              SEARCH_RULES*
SEG              SEGRUN*          SERVERS*         SPOOL*
SYSCOM           SYSOVL           SYSTEM           SYSTEM_DEBUG*
T&MRUN           TOOLS            TRANSLATORS      UP_LINE_DUMP*


```
Are they missing from the tapes, or am I just being dumb?

**Comment by drboone** [comment]
Not on those tapes.

As an end customer, when you got a tape from Prime, it of course included only the products for which you were licensed.  (VARs and such tended to be able to get more complete sets.)  Since license fees were (tens of) thousands of dollars per product, no one bought stuff they didn't use.

What we have in captivity is tape sets that end customers received, that have survived in usable form, and that we've been lucky enough  to obtain and read.  Unfortunately, the ones I got were pretty minimal, and those are the ones on sysovl.info right now.  Jim obtained several installed system images, and constructed the public emulators from those and the few tape sets available.  He also did some work to copy layered products across versions where that was workable.  The 19.2 PRIMENET is on several of the images, for example.  (There are some version compatibility restrictions.)  The sample images are derived from the public emulators.

I do have one 23.3 tape set with a larger set of layered products that I'm waiting for permission to release.  It includes PRIMENET, EMACS, TCP/IP, NFS, several compilers, and a few other things.

We do have a few layered products mixed into sets of files which are not install tapes.  Trying to sift through those heaps and extract the shareable parts is an ongoing project.  A few such extracts are at https://yagi.h-net.org/prime_software/ (don't have all of them moved to sysovl.info yet).

**Comment by SirNibble** [comment]
Ah!

That makes a lot of sense, cheers.

So at the moment there's still going to be an element of patch-working? That's fair enough, and just adds  to the fun.
